### DISCUSSION TITLE
本地包能否增加版本支持

### DISCUSSION THREAD

**Discussion by wulahaier** [xmake-io/xmake#2658] [Q&A] [answered]
使用xmake package 生成的本地包能否自动读取xmake.lua了里面指定的版本, 并且把版本号保留在生成的本地包里。
这样在下游项目使用本地包时可以和像使用远程包一样使用semantic version。

**Comment by waruqi** [comment, marked as answer]
搞不了，add_versions 是要跟 add_urls 里面的 tar shasum 和 git commit 绑定的，本地包所有 libs/headers 都是分散存储，无法校验 也无法根据版本选择包。。

只有打成 remote 包，才能支持 versions，目前原本就支持

**Reply by wulahaier** [reply]
好的，明白了。
